Happening now

xiongchiamiov opened the pull request:
We want to start featuring live threads on the homepage when something big's
happening. To do that, we need a way of setting it, which is this little form.

Related: #111 

:eyeglasses: @umbrae @spladug

xiongchiamiov committed:
Happening now: add admin control panel

xiongchiamiov committed:
fixup: just use a simple url field

spladug commented on reddit_liveupdate/controllers.py:
```diff
@@ -1090,3 +1092,38 @@ def GET_mediaembed(self, liveupdate, embed_index):
+    )
+    def POST_happening_now(self, featured_thread):
+        if featured_thread:
+            featured_thread = featured_thread._id
```
nit: it's usually cleaner not to rewrite a variable but rather to use a new name.

spladug commented on reddit_liveupdate/controllers.py:
```diff
@@ -1090,3 +1092,38 @@ def GET_mediaembed(self, liveupdate, embed_index):
+        candidate_threads = LiveUpdateEvent._byID(list(candidate_thread_ids),
+                return_dict=False)
+
+        candidate_threads = filter(lambda t: not (t.banned or
```
nit: since this is in public code now, it might be a good time to clean up the indents to match our style guide.

spladug commented on reddit_liveupdate/templates/happeningnowadmin.html:
```diff
@@ -0,0 +1,14 @@
+%if thing.current_thread:
+    <div class="infobar">
+        warning: we're already featuring a live thread. you shouldn't submit a new link unless you are deleting this one:<br />
```
the strings should be translated

spladug commented on reddit_liveupdate/controllers.py:
```diff
@@ -1090,3 +1092,38 @@ def GET_mediaembed(self, liveupdate, embed_index):
+
+
+@add_controller
+class LiveUpdateAdminController(RedditController):
```
I don't feel strongly, but is there any reason for this to have a separate controller and the admin page trappings? I don't see it being injected into the admin menu anywhere so it's just sorta faking being part of the admin UI right?

xiongchiamiov replied:
I was going to inject it into the menu separately, since I didn't really see a way to inject it in from here.

spladug commented:
:nail_care: just for minor stuff. i don't feel strongly. thanks for moving this here!

umbrae commented on reddit_liveupdate/validators.py:
```diff
@@ -28,6 +30,19 @@ def run(self, id):
+            return None
+
+        u = UrlParser(url)
+        id = re.match(r'/live/(\w+)/?', u.path)
```
nit: `id` is a keyword in python so I tend to try to use other variable names when possible.

umbrae replied:
Should we also check `is_reddit_url` here?

xiongchiamiov replied:
Yeah, good idea ( 82c26ac ).  I was going to add some unit tests here, too, but there are some hiccups to getting that working and I figured it wasn't a big deal.

umbrae commented on reddit_liveupdate/controllers.py:
```diff
@@ -1090,3 +1092,38 @@ def GET_mediaembed(self, liveupdate, embed_index):
+class LiveUpdateAdminController(RedditController):
+    @validate(VAdmin())
+    def GET_happening_now(self):
+        candidate_thread_ids = queries.get_active_events()
```
I think we can get rid of `candidate_thread_ids` now that we're just using an input, right?

umbrae commented:
:nail_care: :+1:

xiongchiamiov committed:
fixup: translate text

xiongchiamiov committed:
fixup: two-space template indentation

xiongchiamiov committed:
fixup: prune off now-unneeded code

xiongchiamiov committed:
fixup: change indentation

Although permitted by the styleguide, I didn't like how it looked, and it
pushed us over 80 characters.

xiongchiamiov committed:
fixup: don't overwrite variable

xiongchiamiov committed:
fixup: id -> event_id

xiongchiamiov committed:
fixup: ignore non-reddit urls

xiongchiamiov committed:
Merge branch 'master' into happening-now

xiongchiamiov committed:
fixup: move key into a constant

xiongchiamiov commented:
:haircut:

umbrae commented:
:fish: lgtm!

spladug commented:
:fish:

xiongchiamiov commented:
dbe5a503a76d43994ac29bf46d583aa1ab27477f
